sound/soc/intel/boards/sof_maxim_common.c

```diff
@@ -152,12 +198,21 @@ int max_98373_spk_codec_init(struct snd_soc_pcm_runtime *rtd)
 }
 EXPORT_SYMBOL_NS(max_98373_spk_codec_init, SND_SOC_INTEL_SOF_MAXIM_COMMON);
 
-void max_98373_set_codec_conf(struct snd_soc_card *card)
+void max_98373_dai_link(struct snd_soc_dai_link *link)
+{
+	link->codecs = max_98373_components;
+	link->num_codecs = get_maxim_num_codec();
+	link->init = max_98373_spk_codec_init;
+	link->ops = &max_98373_ops;
+}
+EXPORT_SYMBOL_NS(max_98373_dai_link, SND_SOC_INTEL_SOF_MAXIM_COMMON);
+
+void sof_max98373_codec_conf(struct snd_soc_card *card)
 {
 	card->codec_conf = max_98373_codec_conf;
 	card->num_configs = ARRAY_SIZE(max_98373_codec_conf);
```

Comment:
It seems max98373 always has two amps. If so, why do we need to use `get_maxim_num_codec()` to get codec numbers? Why don't we just set `link->num_codecs = 2` in max_98373_dai_link()?
If `get_maxim_num_codec()` could be 4, shouldn't we use different card->codec_conf and card->num_configs, just like `sof_max98390_codec_conf`?